version/src/lib.rs

```diff
@@ -198,6 +201,12 @@ pub mod scale {
         /// # Errors
         /// Will return error if version is unsupported or if input won't have enough bytes for decoding.
         fn decode_versioned(input: &[u8]) -> Result<Self>;
```

Comment:
why don't we remove this now?

Reply:
Why don't we use `decode_all` instead of `decode` everywhere?

Reply:
Because it's impossible to use `decode_all()` if the structure is nested in the other type. There will be extra bytes for the next thing in the containing structure, but our type will give an error just because there are some bytes left. We can use `decode_all()` only for top-most structures